udata/models/owned.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import logging
+
+from blinker import signal
+
+from mongoengine import NULLIFY, Q, pre_save, post_save
+from mongoengine.fields import ReferenceField
+
+from .queryset import UDataQuerySet
+
+log = logging.getLogger(__name__)
+
+
+class OwnedQuerySet(UDataQuerySet):
+    def owned_by(self, *owners):
+        qs = Q()
+        for owner in owners:
+            qs |= Q(owner=owner) | Q(organization=owner)
+        return self(qs)
+
+
+class Owned(object):
+    '''
+    A mixin to factorize owning behvaior between users and organizations.
+    '''
+    owner = ReferenceField('User', reverse_delete_rule=NULLIFY)
+    organization = ReferenceField('Organization', reverse_delete_rule=NULLIFY)
+
+    on_owner_change = signal('Owned.on_owner_change')
+
+    meta = {
+        'indexes': [
+            'owner',
+            'organization',
+        ],
+        'queryset_class': OwnedQuerySet,
```

Comment:
This is Mongoengine behavior: child class inherit this queryset (at least used in tests), unless this is explicitely overriden.